app.py

```diff
@@ -0,0 +1,591 @@
+                    'Status':'Fail',
+                    'Reason': 'Server Error'
+                }
+        else:
+            payLoad = {
+                'Status':'Fail',
+                'Reason': 'No such Event'
+            }
+        
+    else:
+        payLoad = {
+            'Status':'Fail',
+            'Reason': 'AuthFail'
+        }
+
+    emit('attendence_result', json.dumps(payLoad)) #json is emited now, privately
+    try:
+        if status_:
+            # negative otp means ignore it, wrong entry
+            # here recheck for otp is not needed, so can be removed
+            try:
+                event = Events.query.filter_by(otp=otp_).first()
+                event_name_ = event.event_name
+                admin_name_ = event.admin_name
+                attendence_status = 'Success' # This param is for admin only
+            except:
+                event_name_ = ''
+                admin_name_ = ''
+                otp_ = -1
+                attendence_status = 'Failure' # This param is for admin only
+            
+            payLoad = {
+                'email':user_info(token_).get("email"),
+                'username':user_info(token_).get("username"),
+                'datetime':str(datetime_),
+                'status': attendence_status,
+                'event_name':event_name_,
+                'admin_name':admin_name_,
+                'otp':otp_
+            }
+            emit('admin_listen', json.dumps(payLoad), namespace='/admin_namespace')
+        else:
+            raise ValueError('Status failure')
+    except:
+        payLoad = {
+            'email':'',
+            'username':'',
+            'datetime':str(datetime.datetime.now()),
+            'status': 'Failure',
+            'event_name':'',
+            'admin_name':'',
+            'otp':-1
+        }
+        emit('admin_listen', json.dumps(payLoad), namespace='/admin_namespace')
+
+
+
+if __name__ == '__main__':
+    socketio.run(app)
+
+
+
+## Do this:
+# when a new event is created, event-name (slug) is added in events db. Now when attende sends otp we take it and check if it matches with any \
+# event and event attendence is going on. Then we take attendence and emit this info to admin namespace.
+# also, attende attendence is sended to one more route that attendence is emited to that particular event room (if allowed for so by admin).
+
+# attendence is emitted to: 1) attendence namespace, from their attendence is done and emited to admin.
+# 2) broadcaster namespace, from there to specific room name if allowed by admin.
+
+# admin_namespace -> to emit to the admin_namespace :emits to the client side namespace then  (SERVER SIDE NOT NEEDED)
+# attendence_namespace-> attendence (to emit client attendence) :emits to the client side admin_namespace then
+# broadcaster_namespace -> attendence comes to this namespace also, from here it's sended to particular room
+
+
+
+### SocketsIO how and what
+# User can only emit
```

Comment:
You can explain these things in the readme!

Reply:
Ok when i refactor the whole code will make a helpers guide and place their